addons/dynamic-mount/ossfs/docker/mount_and_umount.py

```diff
@@ -0,0 +1,150 @@
+    cmd = f"cat /proc/self/mountinfo | grep fuse | grep {mnt_point}"
+    proc = subprocess.Popen(cmd, shell=True, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+    output, error = proc.communicate()
+    if error:
+        raise Exception(f"Failed to execute command: {cmd}, error message: {error}")
+    return bool(output)
+    
+
+def mount_ossfs(bucket_path, mnt_point, ossfs_options):
+    if not os.path.exists(mnt_point):
+        # create dirs is not exist
+        os.makedirs(mnt_point)
+    if is_fuse_mount(mnt_point):
+        # if mnt_point is fuse mountpoint, pass mount
+        print(f"{mnt_point} is a fuse mount, skip mount")
+        return
+    retry_count = 3
+    while retry_count:
+        try:
+            print("mount_ossfs: ossfs", bucket_path, mnt_point, ossfs_options)
+            subprocess.check_call(['ossfs', bucket_path, mnt_point] + ossfs_options.split())
+            return
+        except subprocess.CalledProcessError as e:
+            print(f"Failed to mount, error: {e}")
+            retry_count -= 1
+            if retry_count:
+                print("retry...")
+                time.sleep(1) 
+            else:
+                raise 
+
+def umount_ossfs(mnt_point):
+    if not os.path.exists(mnt_point):
+        # pass not exist mnt_point
+        print(f"{mnt_point} does not exist")
+        return
+
+    if not is_fuse_mount(mnt_point):
+        # if mnt_point is not fuse mountpoint, remove
+        print(f"{mnt_point} is not a FUSE mount point")
+        try:
+            shutil.rmtree(mnt_point)
+        except OSError as e:
+            print(f"Failed to remove {mnt_point}, error: {e}")
+        return
+
+    try:
+        print("umount_ossfs: umount", mnt_point)
+        subprocess.check_call(['umount', mnt_point])
+    except subprocess.CalledProcessError as e:
+        print(f"Failed to unmount {mnt_point}, error: {e}")
+        return
+    
+    try:
+        shutil.rmtree(mnt_point)
+    except OSError as e:
+        print(f"Failed to remove {mnt_point}, error: {e}")
+
+if __name__=="__main__":
+    rawStr = ""
+    with open("/etc/fluid/config/config.json", "r") as f:
+        rawStr = f.readlines()
+
+    print(rawStr[0])
+
+    obj = json.loads(rawStr[0])
+
+    if len(sys.argv) > 1:
+        mounted_path = sys.argv[1:]
+    else:
+        mounted_path = []
+    mounted = []
+    for p in mounted_path:
+        if not p.startswith("/"):
+            continue
+        mounted.append(p.split("/")[-1])
+    target_mounted = [mount["name"] for mount in obj["mounts"]]
+
+    need_mount = list(set(target_mounted).difference(set(mounted)))
+    need_unmount = list(set(mounted).difference(set(target_mounted)))
+    print(f"need mount: {need_mount}, need umount: {need_unmount}")
+
+    for mount in obj["mounts"]:
+        if mount["name"] not in need_mount:
+            continue
+        bucket = mount["mountPoint"].lstrip("oss://")
+        bucket_path = bucket
+        path = mount.get("path")
```

Comment:
`path` here has different meaning with `Dataset.spec.mounts[*].path`. `Dataset.spec.mounts[*].path` means where to mount the `Dataset.spec.mounts[*].mountPoint`. 

The correct mount command should be:
```
ossfs <bucket>:<bucket_path> /runtime-mnt/thin/default/thin-demo/thin-fuse/<path>
```
where both <bucket> and <bucket_path> should be extracted from `Dataset.spec.mounts[*].mountPoint`.

Reply:
Fixed.